language/def_spec.rb

```diff
@@ -630,5 +630,77 @@ def some_method; end
 
       DefSpecsLambdaVisibility.should have_private_instance_method("some_method")
     end
+    
+    it "creates instance methods using instance_eval on an anon class" do
+      m = Class.new
+      m.instance_eval do
+        def klass_method
+          :test
+        end
+      end
+      
+      m.klass_method.should == :test
+    end
+    
+    it "creates instance methods using instance_exec on an anon class" do
+      m = Class.new
+      m.instance_exec(1) do |param|
+        @stuff = param
+        
+        def klass_method
+          @stuff
+        end
+      end
+      
+      m.klass_method.should == 1
+    end
+    
+    it "creates instance methods using instance_eval inside an anon class" do
+      m = Class.new do
+        instance_eval do
+          def klass_method
+            :test
+          end          
+        end
+      end
+      
+      m.klass_method.should == :test
+    end
+    
+    it "creates instance methods using instance_exec inside an anon class" do
+      m = Class.new do
+        instance_exec(2) do |param|
+          @stuff = param
+          
+          def klass_method
+            @stuff
+          end          
+        end
+      end
+      
+      m.klass_method.should == 2
+    end
```

Comment:
That makes sense, except Opal's compiler and runtime do things differently based on whether you def a method inside a block vs. outside. As a result, I think a separate spec might be needed since this cannot be tested with Class.new alone.